sklego/preprocessing/pandastransformers.py

```diff
@@ -1,11 +1,60 @@
+from __future__ import annotations
+
 import narwhals as nw
-import pandas as pd
+from narwhals.dependencies import get_pandas
 from sklearn.base import BaseEstimator, TransformerMixin
 from sklearn.utils.validation import check_is_fitted
 
 from sklego.common import as_list
 
 
+def _nw_match_dtype(dtype, selection):
+    if selection == "number":
+        return dtype in (
+            nw.Int64,
+            nw.Int32,
+            nw.Int16,
+            nw.Int8,
+            nw.UInt64,
+            nw.UInt32,
+            nw.UInt16,
+            nw.UInt8,
+            nw.Float64,
+            nw.Float32,
+        )
+    if selection == "bool":
+        return dtype == nw.Boolean
+    if selection == "string":
+        return dtype == nw.String
+    if selection == "category":
+        return dtype == nw.Categorical
+    msg = f"Expected {{'number', 'bool', 'string', 'category'}}, got: {selection}, which is not (yet!) supported."
+    raise ValueError(msg)
+
+
+def _nw_select_dtypes(df, include: str | list[str], exclude: str | list[str]):
+    feature_names = []
+    if isinstance(include, str):
+        include = [include]
+    if isinstance(exclude, str):
+        exclude = [exclude]
+    for name, dtype in df.schema.items():
+        if include and exclude:
+            if any(_nw_match_dtype(dtype, _include) for _include in include) and not any(
+                _nw_match_dtype(dtype, _exclude) for _exclude in exclude
+            ):
+                feature_names.append(name)
+        elif include:
+            if any(_nw_match_dtype(dtype, _include) for _include in include):
+                feature_names.append(name)
+        elif exclude:
+            if not any(_nw_match_dtype(dtype, _exclude) for _exclude in exclude):
+                feature_names.append(name)
+        else:
+            raise ValueError("Must provide at least one of `include` or `exclude`")
+    return df.select(feature_names)
```

Comment:
Maybe it is _too edgy_ of a trick but I find it particularly pleasing:

```py
```

```suggestion
def _nw_select_dtypes(df, include: str | list[str], exclude: str | list[str]):
    if not include and not exclude:
        raise ValueError("Must provide at least one of `include` or `exclude`")

    if isinstance(include, str):
        include = [include]
    if isinstance(exclude, str):
        exclude = [exclude]

    include = include or ["string", "number", "bool", "category"]
    exclude = exclude or []

    feature_names = [
        name
        for name, dtype in df.schema.items()
        if any(_nw_match_dtype(dtype, _include) for _include in include)
        and not any(_nw_match_dtype(dtype, _exclude) for _exclude in exclude)
    ]
    return df.select(feature_names)
```

Reply:
nice, thanks!